src/execution/BucketExecutor.tsx

```diff
@@ -0,0 +1,338 @@
+        if (this.hasUnmounted) {
+            // tslint:disable-next-line:no-console
+            console.warn("You are calling a component that is already unmounted");
+        }
+        if (!response) {
+            this.getResponse(true, paging);
+        } else {
+            this.getResult(response, paging);
+        }
+    };
+
+    public getResponse = (
+        triggerGetResult: boolean = this.props.autoLoadFirstPage,
+        resultPaging = this.getInitialPaging(),
+    ) => {
+        const { dimensions, filters, sortBy, projectId } = this.state;
+        const execution = getExecutionFromDimensions(dimensions, filters, sortBy);
+        const responsePromise = this.sdk.execution.getExecutionResponse(projectId, execution);
+        this.setState({
+            result: null,
+            response: null,
+            isLoading: true,
+            error: null,
+            responsePromise,
+        });
+        responsePromise.then(
+            response => {
+                if (responsePromise !== this.state.responsePromise || this.hasUnmounted) {
+                    // tslint:disable-next-line:no-console
+                    console.warn("Discarding outdated response promise", responsePromise);
+                    return;
+                }
+                if (triggerGetResult) {
+                    this.setState({ response, responsePromise: null });
+                    this.getResult(response, resultPaging);
+                } else {
+                    this.setState({ response, isLoading: false, responsePromise: null });
+                }
+            },
+            error => {
+                if (responsePromise !== this.state.responsePromise || this.hasUnmounted) {
+                    // tslint:disable-next-line:no-console
+                    console.warn("Discarding outdated response promise", responsePromise);
+                    return;
+                }
+                this.setState({ error: convertErrors(error), isLoading: false, responsePromise: null });
+            },
+        );
+    };
+
+    public getResult = (response: Execution.IExecutionResponse, paging: IPaging) => {
+        const resultPromise = this.sdk.execution.getPartialExecutionResult(
+            response.links.executionResult,
+            paging.limit,
+            paging.offset,
+        );
+        this.setState({ isLoading: true, result: null, error: null, resultPromise });
+        resultPromise.then(
+            result => {
+                if (resultPromise !== this.state.resultPromise || this.hasUnmounted) {
+                    // tslint:disable-next-line:no-console
+                    console.warn("Discarding outdated result promise", resultPromise);
+                    return;
+                }
+                this.setState({ result, isLoading: false, resultPromise: null });
+            },
+            error => {
+                if (resultPromise !== this.state.resultPromise || this.hasUnmounted) {
```

Comment:
This `if` is duplicated 4 times in this file (twice for result, twice for response). Please consider making a method out of this.

Reply:
Its actually a bit different an awkward to extract